analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -151,7 +154,42 @@ IEnumerable<TMessage> ReadProtobuf()
         }
 
         public IEnumerable<Compilation> Compile(bool concurrentAnalysis) =>
-            CreateProject(concurrentAnalysis).Solution.Compile(builder.ParseOptions.ToArray());
+            builder.IsRazor ? CompileRazor() : CreateProject(concurrentAnalysis).Solution.Compile(builder.ParseOptions.ToArray());
+
+        private IEnumerable<Compilation> CompileRazor()
+        {
+            if (!MSBuildLocator.IsRegistered)
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
+
+            using var workspace = MSBuildWorkspace.Create();
+            workspace.WorkspaceFailed += (_, failure) => Console.WriteLine(failure.Diagnostic);
+
+            // Copy razor project directory and test case files to a temporary build location
+            var tempPath = Path.Combine(Path.GetTempPath(), $"ut-razor-{Guid.NewGuid()}");
+            try
+            {
+                Directory.CreateDirectory(tempPath);
+
+                foreach (var file in Directory.GetFiles("TestFramework\\Razor\\EmptyProject").Concat(builder.Paths.Select(TestCasePath)))
+                {
+                    File.Copy(file, Path.Combine(tempPath, Path.GetFileName(file)));
+                }
+
+                return new[]
+                {
+                    workspace.OpenProjectAsync(Path.Combine(tempPath, "EmptyProject.csproj")).Result.GetCompilationAsync().Result
+                };
```

Comment:
NITPICK:
```suggestion
                return new[] { workspace.OpenProjectAsync(Path.Combine(tempPath, "EmptyProject.csproj")).Result.GetCompilationAsync().Result };
```

Reply:
Since I will work on top of this infrastructure anyway, I can address this in a follow-up PR. I will merge as is and address it later.